packages/lib/ren-tx/src/configs/genericBurn.ts

```diff
@@ -113,128 +106,175 @@ const burnTransactionListener = (context: BurnMachineContext) => (
-                    releaseRef.removeListener(
-                        "transaction",
-                        transactionListener,
-                    );
-                    releaseRef.removeListener("txHash", hashListener);
-                });
-                releaseRef.catch((e) => {
-                    console.error("release error", e);
-                    callback({ type: "RELEASE_ERROR", data: new Error(e) });
+                    burnRef._cancel();
+                    burnRef.removeListener("confirmation", burnListener);
                 });
 
                 try {
-                    const res = await releaseRef
-                        .on("status", releaseListener)
-                        .on("transaction", transactionListener)
-                        .on("txHash", hashListener);
-                    callback({
-                        type: "RELEASED",
-                        data: { ...tx, renResponse: res },
-                    });
-                } catch (e) {
-                    callback({ type: "RELEASE_ERROR", data: new Error(e) });
+                    const r = await burnRef
+                        // host chain tx hash
+                        .on("transactionHash", (txHash: string) => {
+                            tx = {
+                                sourceTxHash: txHash,
+                                sourceTxConfs: 0,
+                                sourceTxAmount: Number(
+                                    context.tx.suggestedAmount,
+                                ),
+                                rawSourceTx: {
+                                    amount: String(context.tx.suggestedAmount),
+                                    transaction: {},
+                                },
+                            };
+                            callback({
+                                type: "SUBMITTED",
+                                data: {
+                                    ...tx,
+                                },
+                            });
+                        })
+                        .on("confirmation", burnListener);
+                    if (tx) {
+                        // the burn status is canonical, we should proceed if
+                        // it says we have burned
+                        if (r.status == BurnAndReleaseStatus.Burned) {
+                            // FIXME: ensure that the confs always match
+                            // the target
+                            tx.sourceTxConfs = 7;
+                        }
+
+                        // Always call because we won't get an emission
+                        // if the burn is already done
+                        burnListener(tx.sourceTxConfs);
+                    }
+                } catch (error) {
+                    throw error;
                 }
             };
 
-            receive((event: BurnMachineEvent) => {
+            receive(async (event: BurnMachineEvent) => {
```

Comment:
Does the body of this function need to be in a try-catch to avoid `Unhandled promise rejection` warnings or is that handled by the receiver?

Reply:
Good point - this might actually have been the cause for some breakage experienced at other levels, will address

Reply:
Addressed